packages/components/src/combobox-control/index.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * External dependencies
+ */
+import { useCombobox } from 'downshift';
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import { Button, Dashicon } from '../';
+
+const itemToString = ( item ) => item && item.name;
+export default function ComboboxControl( {
+	className,
+	hideLabelFromVision,
+	label,
+	options: items,
+	onInputValueChange: onInputValueChange,
+	onChange: onSelectedItemChange,
+	value: _selectedItem,
+} ) {
+	const {
+		getLabelProps,
+		getToggleButtonProps,
+		getComboboxProps,
+		getInputProps,
+		getMenuProps,
+		getItemProps,
+		isOpen,
+		highlightedIndex,
+		selectedItem,
+	} = useCombobox( {
+		initialSelectedItem: items[ 0 ],
+		items,
+		itemToString,
+		onInputValueChange,
+		onSelectedItemChange,
+		selectedItem: _selectedItem,
+	} );
+	const menuProps = getMenuProps( {
+		className: 'components-combobox-control__menu',
+	} );
+	// We need this here, because the null active descendant is not
+	// fully ARIA compliant.
+	if (
+		menuProps[ 'aria-activedescendant' ] &&
+		menuProps[ 'aria-activedescendant' ].slice(
+			0,
+			'downshift-null'.length
+		) === 'downshift-null'
+	) {
+		delete menuProps[ 'aria-activedescendant' ];
+	}
+	return (
+		<div
+			className={ classnames( 'components-combobox-control', className ) }
+		>
+			{ /* eslint-disable-next-line jsx-a11y/label-has-associated-control, jsx-a11y/label-has-for */ }
+			<label
+				{ ...getLabelProps( {
+					className: classnames(
+						'components-combobox-control__label',
+						{
+							'screen-reader-text': hideLabelFromVision,
+						}
+					),
+				} ) }
+			>
+				{ label }
+			</label>
+			<div
+				{ ...getComboboxProps( {
+					className: 'components-combobox-control__button',
+				} ) }
+			>
+				<input
+					{ ...getInputProps( {
+						className: 'components-combobox-control__button-input',
+					} ) }
+				/>
+				<Button
+					{ ...getToggleButtonProps( {
+						// This is needed because some speech recognition software don't support `aria-labelledby`.
+						'aria-label': label,
+						'aria-labelledby': undefined,
+						className: 'components-combobox-control__button-button',
+					} ) }
+				>
+					<Dashicon
+						icon="arrow-down-alt2"
+						className="components-combobox-control__button-icon"
+					/>
+				</Button>
+			</div>
+			<ul { ...menuProps }>
+				{ isOpen &&
```

Comment:
Can we check that the input has a non-empty value before rendering the items? I'm still seeing the bug where, after typing in a letter and then deleting it so the input becomes empty, the whole list shows. Or perhaps this should be fixed in Downshift instead? Cc @silviuaavram

Reply:
`isOpen` does not take into account the value in `inputValue`, so you can have the menu open without having anything in the input. Is there an ARIA/general agreed behavior for this? I am not aware of it.

To customise you can probably use `stateReducer` and only make `isOpen` true if `inputValue` has content. If you consider the default behavior as a bug, can you open an issue in Downshift and have it documented? Our hooks are very easy to customise behavior, but we should have the standard actions on par with ARIA guidelines.

Thanks!

Reply:
Yup, that's the behaviour in the official [ARIA combobox examples](https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html). Check the third example, that has the button just like ours. The whole menu is only visible on click of the button, or on Arrow Down keypress. But if you type in the box and then clear, the dropdown disappears.